plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -338,44 +329,40 @@ export class ActionMatcher {
     /**
      * Sublevel 4 of action matching.
      */
-    private checkEventAgainstPersonFilter(
-        personProperties: Properties | undefined,
-        filter: PersonPropertyFilter
-    ): boolean {
-        if (!personProperties) {
+    private checkEventAgainstPersonFilter(event: PostIngestionEvent, filter: PersonPropertyFilter): boolean {
+        if (!event.person_properties) {
             return !!(filter.operator && emptyMatchingOperator[filter.operator]) // NO PERSON OR PROPERTIES TO MATCH AGAINST FILTER
         }
-        return this.checkPropertiesAgainstFilter(personProperties, filter)
+        return this.checkPropertiesAgainstFilter(event.person_properties, filter)
     }
 
     /**
      * Sublevel 4 of action matching.
      */
-    private checkEventAgainstElementFilter(elements: Element[], filter: ElementPropertyFilter): boolean {
+    private checkEventAgainstElementFilter(event: PostIngestionEvent, filter: ElementPropertyFilter): boolean {
         if (filter.key === 'selector') {
             const okValues = Array.isArray(filter.value) ? filter.value : [filter.value]
             return okValues.some((okValue) =>
-                okValue ? this.checkElementsAgainstSelector(elements, okValue.toString()) : false
+                okValue ? this.checkElementsAgainstSelector(this.getElementsList(event), okValue.toString()) : false
             )
         } else {
-            return elements.some((element) => this.checkPropertiesAgainstFilter(element, filter))
+            return this.getElementsList(event).some((element) => this.checkPropertiesAgainstFilter(element, filter))
         }
     }
 
     /**
      * Sublevel 4 of action matching.
      */
     private async checkEventAgainstCohortFilter(
-        personUuid: string | undefined,
-        teamId: number,
+        event: PostIngestionEvent,
         filter: CohortPropertyFilter
     ): Promise<boolean> {
         let cohortId = filter.value
         if (cohortId === 'all') {
             // The "All users" cohort matches anyone
             return true
         }
-        if (!personUuid) {
+        if (!event.person_id) {
             return false // NO PERSON TO MATCH AGAINST COHORT
         }
```

Comment:
It still is optional from what I can see? Unless `RawClickHouseEvent` and in turn `BaseEvent` shouldn't be optional? Either way that feels like not part of this PR

Reply:
ok lets leave it as is (BaseEvent is only used for `ClickHouseEvent` and `RawClickHouseEvent` which both always have person info), just noticed it because other type changes were made, but indeed a separate PR would be good.